config/kernel-vfs-iterate.m4

```diff
@@ -23,16 +23,27 @@ AC_DEFUN([ZFS_AC_KERNEL_VFS_ITERATE], [
 		dnl #
 		dnl # 3.11 API change
 		dnl #
+		dnl # RHEL 7.5 compatibility; the fops.iterate() method was
+		dnl # added to the file_operations structure but in order to
+		dnl # maintain KABI compatibility all callers must set
+		dnl # FMODE_KABI_ITERATE which is checked in iterate_dir().
+		dnl # When detected ignore this interface and fallback to
+		dnl # to using fops.readdir() to retain KABI compatibility.
+		dnl #
 		AC_MSG_CHECKING([whether fops->iterate() is available])
 		ZFS_LINUX_TRY_COMPILE([
 			#include <linux/fs.h>
-			int iterate(struct file *filp, struct dir_context * context)
-			    { return 0; }
+			int iterate(struct file *filp,
+			    struct dir_context *context) { return 0; }
 
 			static const struct file_operations fops
 			    __attribute__ ((unused)) = {
 				.iterate	 = iterate,
 			};
+
+			//#if defined(FMODE_KABI_ITERATE)
+			//#error "RHEL 7.5, FMODE_KABI_ITERATE interface"
+			//#endif
```

Comment:
As described in the comment, this interface is intentionally not used in order to maintain KABI compatibility.  What isn't mentioned in the comment, is that while the interface was added, it isn't used by any of the major Linux filesystems in the RHEL 7.5 kernel.  Since it isn't as well tested in this kernel we intentionally choose to continue using the old interface.  I'd suggest you do the same in this patch.